cohortextractor/query_engines/base_sql.py

```diff
@@ -59,10 +59,6 @@ class BaseSQLQueryEngine(BaseQueryEngine):
     # No limit by default although some DBMSs may impose one
     max_rows_per_insert: Optional[int] = None
 
-    def __init_subclass__(cls, **kwargs):
-        super().__init_subclass__(**kwargs)
-        assert cls.sqlalchemy_dialect is not None
```

Comment:
Did you test this is still enforced? I found with the initial work I did we need to make use of a Protocol to enforce a value on the subclass is defined, but we're currently using class types which don't enforce this at instantiation time.

Reply:
I didn't, but I sort of felt I didn't care that much! My feeling is that if we need to add `__init_subclass` for other reasons then it's reasonable to assert that things are as we expect them to be before we do the initialisation. But adding an `__init_subclass` method _just_ for this purpose seems overkill. Especially as, in this case, it's not like it's preventing a subtle error that you might not otherwise notice :)

Reply:
I think we've been going _slightly_ overboard with asserts on things like this, so I have no problem seeing it going. Asserts can be really helpful for more complex constraints or those that would cause weird problems if they are broken -- but as Dave suggests this will be really obvious if it happens.